src/main/java/step4/Cars.java

```diff
@@ -0,0 +1,47 @@
+package step4;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class Cars {
+
+    private static final String RANKED_STRING_DELIMITER = ",";
+
+    private List<Car> cars = new ArrayList<>();
+
+    public Cars(List<Car> cars) {
+        this.cars = cars;
+    }
+
+    public void setupCars(String[] carsNames, CarMoveStrategy carMoveStrategy) {
+        Arrays.stream(carsNames).forEachOrdered(name ->
+                cars.add(new Car(name, 0, carMoveStrategy)));
+    }
+
+    public String getWinners() {
+
+        Optional<Car> max = cars.stream().max((o1, o2) -> (o1.getLocation() >= o2.getLocation()) ? 1 : -1);
+        Stream<Car> ranked = cars.stream().filter(o -> o.getLocation() == max.get().getLocation());
+        String winners = ranked.map(Car::getCarName).collect(Collectors.joining(RANKED_STRING_DELIMITER));
```

Comment:
Optional과 Stream 사용 잘 해주셨습니다 👍 
조금 다듬을 수 있을 것 같은데요.
```suggestion
        int maxLocation = cars.stream().map(Car::getLocation).max(Integer::compare).orElse(0); // 기본 값은 최소 위치
        return cars.stream().filter(car -> car.getLocation == maxLocation)
                  .collect(Collectors.joining(RANKED_STRING_DELIMITER));
```
(IDE에서 작성한게 아니라 코드 컴파일이 안될 수도 있습니다^^;;)
참고만 해주세요:)

Reply:
네 문자열 부분만 살짝 얹어서 잘 수정 했습니다. ^^